src/routes/guilds/#guild_id/index.ts

```diff
@@ -43,8 +43,8 @@ router.patch("/", check(GuildUpdateSchema), async (req: Request, res: Response)
 	const perms = await getPermission(req.user_id, guild_id);
 	perms.hasThrow("MANAGE_GUILD");
 
-	body.icon = await handleFile(`/icons/${guild_id}`, body.icon);
-	body.banner = await handleFile(`/banners/${guild_id}`, body.banner);
+	if (body.icon) body.icon = await handleFile(`/icons/${guild_id}`, body.icon);
```

Comment:
why do we need to check if its defined?
handleFile will return null if body.icon is undefined

Reply:
It returns null, and that overwrites the current icon and banner. We want to be able to update the guild settings without providing every single parameter.